azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -243,14 +217,11 @@ public Project getProject(String name) {
      */
     public Project getProject(int id) {
         Project fetchedProject = null;
-        if (isActiveProject(id)) {
-            fetchedProject = projectsById.get(id);
-        } else {
-            try {
-                fetchedProject = projectLoader.fetchProjectById(id);
-            } catch (ProjectManagerException e) {
-                logger.error("Could not load project from store.", e);
-            }
+        try {
+            fetchedProject = projectLoader.fetchProjectById(id);
+            loadAllProjectFlows(fetchedProject);
```

Comment:
loadAllProjectFlows() will set the Map<string, Flow> flows for the project. getProject() is called in so many places, not able to enumerate them all...Most of them (>90%) will need to get the flow info, probably not all.